packages/aws-rfdk/lib/core/lib/utils.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export class Utils {
+  /**
+   * This method compares two version strings
+   *
+   * @param versionFirst
+   * @param versionSecond
+   *
+   * @returns negative value if first version is smaller than second version;
+   * 0 if the versions matches, positive value if first version is smaller
+   * than second version.
+   */
+  public static versionCompare(versionFirst: string, versionSecond: string): number {
+    const regExStripZero = /(\.0+)+$/;
```

Comment:
I wonder if we should handle cases with a trailing dot: `10.1.0.`.
Not sure if we need it, but something like `(\.0+.+)+$` will do the trick.

What do you think?

Reply:
The method 'versionCompare' assumes that the version strings are already formatted. I added another validation for the version format. So, if the version string itself is incorrect, we will throw an error accordingly. This method will continue to just compare the formatted version strings.